backend/compact-connect/lambdas/purchases/data_model/schema/compact.py

```diff
@@ -0,0 +1,58 @@
+# pylint: disable=invalid-name
+from marshmallow import pre_dump, Schema
+from marshmallow.fields import String, Decimal, Nested, List
+from marshmallow.validate import Length, OneOf
+
+from config import config
+from data_model.schema.base_record import BaseRecordSchema, ForgivingSchema
+
+COMPACT_TYPE = 'compact'
+
+class CompactCommissionFeeSchema(Schema):
+    feeType = String(required=False, allow_none=False, validate=OneOf(['FLAT_RATE']))
+    feeAmount = Decimal(required=False, allow_none=False)
+
+@BaseRecordSchema.register_schema(COMPACT_TYPE)
+class CompactRecordSchema(BaseRecordSchema):
+    """
+    Schema for the root compact configuration records
+    """
+    _record_type = COMPACT_TYPE
+
+    # Provided fields
+    compactName = String(required=True, allow_none=False, validate=OneOf(config.compacts))
+    compactCommissionFee = Nested(CompactCommissionFeeSchema(), required=False, allow_none=False)
```

Comment:
Looks like this is _not_ required then? If that's true, can we update the README?

Reply:
Good catch. I initially left many of these fields optional as I was adding tests for the endpoint, but this should definitely be required.